tests/test_flows.py

```diff
@@ -735,13 +736,44 @@ def test_link_metric_query(self):
                                            env.agents[0].mac, env.controller.mac,
                                            mid)
 
-        debug("Checking response contains both Tx and Rx metrics")
-        assert self.check_cmdu_has_tlvs(resp, 0x09)
-        assert self.check_cmdu_has_tlvs(resp, 0x0a)
-
-        # TODO: we can't verify each reported link because
-        # TODO: tshark reports all tlv types using same key
-        # TODO: postponing test implementation until this is solved
+        debug("Checking response contains expected links to/from agent and nothing else")
+        # neighbor macs are based on the setup in "launch_environment_docker" method
+        expected_tx_link_neighbors = [env.controller.mac, env.agents[1].mac]
+        expected_rx_link_neighbors = [env.controller.mac, env.agents[1].mac]
+        actual_tx_links = self.check_cmdu_has_tlvs(resp, 0x09)
+        actual_rx_links = self.check_cmdu_has_tlvs(resp, 0x0a)
+
+        def verify_links(links, expected_neighbors, link_type):
+            verified_neighbors = []
+            unexpected_neighbors = []
+            for link in links:
+                self.safe_check_obj_attribute(link, 'responder_mac_addr', env.agents[0].mac,
+                                              "Responder MAC address is wrong")
+
+                try:
+                    # a tshark (v3.2.4) bug causes "neighbor_mac_addr" field to
+                    # show up as "responder_mac_addr_2"
+                    if link.responder_mac_addr_2 in expected_neighbors:
+                        verified_neighbors += [link.responder_mac_addr_2]
+                    else:
+                        unexpected_neighbors += [link.responder_mac_addr_2]
+                except AttributeError as e:
+                    self.fail(e)
```

Comment:
Shouldn't we also try `neighbor_mac_addr` if `responder_mac_addr_2` doesn't exist (i.e. if we use a different tshark version)?

Reply:
I read the wireshark code and I suspect all Wireshark versions have the same problem. Also the other variables are always filled with zeros. I'll double check with EasyMesh spec but I decided not to use them until I did so.